python/cuvs_bench/cuvs_bench/run/__main__.py

```diff
@@ -0,0 +1,216 @@
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import os
+from pathlib import Path
+from typing import Optional
+
+import click
+from run import run_benchmark
+
+
+@click.command()
+@click.option(
+    "--subset-size",
+    type=click.IntRange(min=1),
+    help="The number of subset rows of the dataset to build the index",
+)
+@click.option(
+    "-k",
+    "--count",
+    default=10,
+    show_default=True,
+    type=click.IntRange(min=1),
+    prompt="Enter the number of  neighbors to search for",
+    help="The number of nearest neighbors to search for",
+)
+@click.option(
+    "-bs",
+    "--batch-size",
+    default=10000,
+    show_default=True,
+    type=click.IntRange(min=1),
+    prompt="Enter the batch size",
+    help="Number of query vectors to use in each query trial",
+)
+@click.option(
+    "--dataset-configuration",
+    default=None,
+    show_default=True,
+    help="Path to YAML configuration file for datasets",
+)
+@click.option(
+    "--configuration",
+    help="Path to YAML configuration file or directory for algorithms. "
+    "Any run groups found in the specified file/directory will "
+    "automatically override groups of the same name present in the "
+    "default configurations, including `base`.",
+)
+@click.option(
+    "--dataset",
+    default="glove-100-inner",
+    show_default=True,
+    prompt="Enter the name of dataset",
+    help="Name of dataset",
+)
+@click.option(
+    "--dataset-path",
+    default=lambda: os.environ.get(
+        "RAPIDS_DATASET_ROOT_DIR",
+        os.path.join(Path(__file__).parent, "datasets/"),
+    ),
+    show_default=True,
+    prompt="Enter the path to dataset folder",
+    help="Path to dataset folder, by default will look in "
+    "RAPIDS_DATASET_ROOT_DIR if defined, otherwise a datasets "
+    "subdirectory from the calling directory.",
+)
+@click.option("--build", is_flag=True, help="Build the index")
+@click.option("--search", is_flag=True, help="Perform the search")
+@click.option(
+    "--algorithms",
+    default="cuvs_cagra",
+    show_default=True,
+    prompt="Enter the comma separated list of named algorithms to run",
+    help="Run only comma separated list of named algorithms. If parameters "
+    "`groups` and `algo-groups` are both undefined, then group `base` "
+    "is run by default.",
+)
+@click.option(
+    "--groups",
+    default="base",
+    show_default=True,
+    prompt="Enter the comma separated groups of parameters",
+    help="Run only comma separated groups of parameters",
+)
+@click.option(
+    "--algo-groups",
+    help="Add comma separated <algorithm>.<group> to run. Example usage: "
+    ' "--algo-groups=raft_cagra.large,hnswlib.large".',
```

Comment:
Does this need to be updated to cuvs?
```suggestion
    ' "--algo-groups=cuvs_cagra.large,hnswlib.large".',
```